nick_work/fdaaa_port.py

```diff
@@ -0,0 +1,315 @@
+        is_covered_phase(phase) and
+        is_not_device_feasibility(primary_purpose) and
+        start_date >= effective_date and
+        is_not_withdrawn(study_status)):
+        act_flag = True
+    else:
+        act_flag = False
+        
+    intervention_type = soup.find_all("intervention_type")
+    locs = t(soup.location_countries)
+
+    if (is_interventional(study_type) and
+        is_covered_intervention(intervention_type) and
+        is_covered_phase(phase) and
+        is_not_device_feasibility(primary_purpose) and
+        available_completion_date >= effective_date and
+        start_date < effective_date and
+        is_not_withdrawn(study_status) and
+        (is_fda_reg(fda_reg_drug, fda_reg_device) or
+         is_old_fda_regulated(is_fda_regulated, fda_reg_drug, fda_reg_device)) and
+        has_us_loc(locs)):
+        old_pact_flag = True
+    else:
+        old_pact_flag = False
+
+    if (is_interventional(study_type) and
+        is_fda_reg(fda_reg_drug, fda_reg_device) and
+        is_covered_phase(phase) and
+        is_not_device_feasibility(primary_purpose) and
+        start_date < effective_date and
+        available_completion_date >= effective_date and
+        is_not_withdrawn(study_status)):
+        new_pact_flag = True
+    else:
+        new_pact_flag = False
+
+    if old_pact_flag == True or new_pact_flag == True:
+        included_pact_flag = True
+    else:
+        included_pact_flag = False
+        
+    location = dict_or_none(parsed_json,[cs, 'location_countries'])
+
+    has_results = does_it_exist(soup.results_first_submitted)
+        
+    pending_results = does_it_exist(soup.pending_results)
+
+
+    pending_data = dict_or_none(parsed_json,[cs,'pending_results'])
+
+    if ((act_flag == True or included_pact_flag == True) and
+        date.today() > available_completion_date + relativedelta(years=1) + timedelta(days=30) and
+        (has_certificate == 0 or (date.today() > available_completion_date + relativedelta(years=3) + timedelta(days=30)))):
+        results_due = True
+    else:
+        results_due = False 
+    
+    results_submitted_date = (str_to_date(soup.results_first_submitted))[0]
+    
+    last_updated_date = (str_to_date(soup.last_update_submitted))[0]
+    
+    certificate_date = (str_to_date(soup.disposition_first_submitted))[0]
+
+    enrollment = t(soup.enrollment)
+
+    sponsor = t(soup.sponsors.lead_sponsor.agency)
+
+    sponsor_type = t(soup.sponsors.lead_sponsor.agency_class)
+
+    collaborators = dict_or_none(parsed_json, [cs, 'sponsors', 'collaborator'])  
+
+    exported = t(soup.oversight_info.is_us_export)
+        
+    url = (soup.url).text
+    
+    official_title = t(soup.official_title)
+
+    brief_title = t(soup.brief_title)
+
+    if official_title is not None:
+        title = official_title
+    elif official_title is None and brief_title is not None:
+        title = brief_title
+    else:
+        title = None
```

Comment:
Remember the `or` lesson of how it yields a component of the items instead of True or False?

`title = official_title or brief_title`

Reply:
Ah, that's really nifty to know!